openpype/tools/workfiles/app.py

```diff
@@ -37,6 +38,182 @@
+
+        template = anatomy.templates[template_key]["file"]
+        if "{comment}" not in template:
+            # Don't look for comment if template doesn't allow it
+            return
+
+        # Create a regex group for extensions
+        extensions = api.registered_host().file_extensions()
+        any_extension = "(?:{})".format(
+            "|".join(re.escape(ext[1:]) for ext in extensions)
+        )
+
+        # Use placeholders that we can safely escape with regex
+        temp_data = copy.deepcopy(data)
+        temp_data["comment"] = "<<comment>>"
+        temp_data["version"] = "<<version>>"
+        temp_data["ext"] = "<<ext>>"
+
+        formatted = anatomy.format(temp_data)
+        fname_pattern = formatted[template_key]["file"]
+        fname_pattern = re.escape(fname_pattern)
+
+        # Replace comment and version with something we can match with
+        # regex
+        fname_pattern = fname_pattern.replace("<<comment>>", "(.+)")
+        fname_pattern = fname_pattern.replace("<<version>>", "[0-9]+")
+        fname_pattern = fname_pattern.replace("<<ext>>", any_extension)
+
+        # Match from beginning to end of string to be safe
+        fname_pattern = "^{}$".format(fname_pattern)
+
+        self.fname_regex = re.compile(fname_pattern)
+
+    def parse_comment(self, filepath):
+        """Parse the {comment} part from a filename"""
+        if not self.fname_regex:
+            return
+
+        fname = os.path.basename(filepath)
+        match = self.fname_regex.match(fname)
+        if match:
+            return match.group(1)
+
+
+class SubversionLineEdit(QtWidgets.QWidget):
+    """QLineEdit with QPushButton for drop down selection of list of strings"""
+    def __init__(self, parent=None):
+        super(SubversionLineEdit, self).__init__(parent=parent)
+
+        layout = QtWidgets.QHBoxLayout(self)
+        layout.setContentsMargins(0, 0, 0, 0)
+        layout.setSpacing(3)
+
+        self._input = PlaceholderLineEdit()
+        self._button = QtWidgets.QPushButton("")
+        self._button.setFixedWidth(18)
+        self._menu = QtWidgets.QMenu(self)
+        self._button.setMenu(self._menu)
+
+        layout.addWidget(self._input)
+        layout.addWidget(self._button)
+
+    @property
+    def input(self):
+        return self._input
+
+    def set_values(self, values):
+        self._update(values)
+
+    def _on_button_clicked(self):
+        self._menu.exec_()
+
+    def _on_action_clicked(self, action):
+        self._input.setText(action.text())
+
+    def _update(self, values):
+        """Create optional predefined subset names
+
+        Args:
+            default_names(list): all predefined names
+
+        Returns:
+             None
+        """
+
+        menu = self._menu
+        button = self._button
+
+        state = any(values)
+        button.setEnabled(state)
+        if state is False:
+            return
+
+        # Include an empty string
+        values = [""] + sorted(values)
```

Comment:
Empty string is just there so you can also quickly clear the field when checking the dropdown menu.